streams/src/test/java/org/apache/kafka/streams/processor/internals/testutil/LogCaptureAppender.java

```diff
@@ -59,6 +85,30 @@ protected void append(final LoggingEvent event) {
         return result;
     }
 
+    public List<Event> getEvents() {
+        final LinkedList<Event> result = new LinkedList<>();
+        synchronized (events) {
+            for (final LoggingEvent event : events) {
+                final String[] throwableStrRep = event.getThrowableStrRep();
+                final Optional<String> throwableString;
+                if (throwableStrRep == null) {
+                    throwableString = Optional.empty();
```

Comment:
nit: maybe we should set a default value instead? If this is empty the test will fail from `NoSuchElementException`.  I know you are aware of this, I'm just thinking about in the future about others using this class and having to test for `isPresent()` vs. `get()`

Reply:
This is exactly what I was going for. The type (`Optional`) documents that you have to check `isPresent` before `get`. It's just as true if we make the field nullable, but then it's not well documented, and therefore more vulnerable to NPEs.

In my test, I deliberately don't check `isPresent` because I expect it to be present. If it's not, we'll get a `NoSuchElementException`, which contains the exact same information as if we had an extra `isPresent` check that then throws an `AssertionException(Expected a throwable)`.

Reply:
If you think it's better to add the `isPresent` assertion, I can do it; I just tend not to do such checks in tests, for the reason above.

Reply:
No, your arguments make sense, I'd leave it as is.